openupgradelib/openupgrade.py

```diff
@@ -643,13 +644,16 @@ def write_value(ids, field, value):
                 cr.execute(query, (params[0], sub_ids))
 
     for model in default_spec.keys():
-        obj = pool.get(model)
-        if not obj:
+        obj = pool.get(model, False)
+        if obj == False:
             do_raise(
                 "Migration: error setting default, no such model: %s" % model)
         for field, value in default_spec[model]:
             domain = not force and [(field, '=', False)] or []
-            ids = obj.search(cr, SUPERUSER_ID, domain)
+            if isinstance(pool, api.Environment):
+                ids = obj.search(domain).ids
```

Comment:
won't all hell break lose later when `write_value` tries to do old style calls again?

Reply:
Yes, the same with _defaults. I didn't tested that yet, but sure it will break.

Reply:
Ok, now I fixed the _defaults, _columns, and the write_value.

What do you think?